### PR TITLE
LPS-87402 filter and sort operations do not work when no accept-language header is set

### PR DESCRIPTION
**Pull request by ruben-pulido**

### CODE REVIEW

**Comment by ruben-pulido** [comment]
ci:test:sf <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by ruben-pulido** [comment]
ci:test:relevant <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/headless-apio/portal/portal-apio-impl/src/main/java/com/liferay/portal/apio/internal/architect/provider/AcceptLanguageProvider.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+			 * 		   {@code Locale}, if the {@code Accept-Language} header is
+			 * 		   present; otherwise returns an empty {@code Stream}
+			 * @review
+			 */
+			@Override
+			public Stream<Locale> getLocales() {
+				String headerAcceptLanguage = httpServletRequest.getHeader(
+					HttpHeaders.ACCEPT_LANGUAGE);
+
+				List<Locale> localesList;
```
Comment (cgoncas) [inline modules/apps/headless-apio/portal/portal-apio-impl/src/main/java/com/liferay/portal/apio/internal/architect/provider/AcceptLanguageProvider.java]: Try not tp use mutable variables.

File: modules/apps/headless-apio/portal/portal-apio-impl/src/main/java/com/liferay/portal/apio/internal/architect/provider/AcceptLanguageProvider.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+			 * 			default locale for the user
+			 * @review
+			 */
+			@Override
+			public Locale getPreferredLocale() {
+				Stream<Locale> localeStream = getLocales();
+
+				Optional<Locale> localeOptional = localeStream.findFirst();
+
+				if (localeOptional.isPresent()) {
```
Comment (cgoncas) [inline modules/apps/headless-apio/portal/portal-apio-impl/src/main/java/com/liferay/portal/apio/internal/architect/provider/AcceptLanguageProvider.java]: Try not to use ifPresent, use map instead for all this method.

File: modules/apps/headless-apio/portal/portal-apio-impl/src/main/java/com/liferay/portal/apio/internal/architect/provider/AcceptLanguageProvider.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+					return localeOptional.get();
+				}
+
+				Optional<CurrentUser> currentUserOptional = Optional.ofNullable(
+					Try.fromFallible(
+						() -> _portal.getUser(httpServletRequest)
+					).map(
+						CurrentUser::new
+					).orElse(
+						null
```
Comment (cgoncas) [inline modules/apps/headless-apio/portal/portal-apio-impl/src/main/java/com/liferay/portal/apio/internal/architect/provider/AcceptLanguageProvider.java]: Never return null

File: modules/apps/headless-apio/portal/portal-apio-impl/src/test/java/com/liferay/portal/apio/internal/architect/provider/AcceptLanguageProviderTest.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+		AcceptLanguageProvider acceptLanguageProvider =
+			new AcceptLanguageProvider();
+
+		MockHttpServletRequest mockHttpServletRequest =
+			new MockHttpServletRequest();
+
+		mockHttpServletRequest.addHeader(
+			HttpHeaders.ACCEPT_LANGUAGE, "someValue");
+
+		List<Locale> locales = new ArrayList<Locale>() {
```
Comment (cgoncas) [inline modules/apps/headless-apio/portal/portal-apio-impl/src/test/java/com/liferay/portal/apio/internal/architect/provider/AcceptLanguageProviderTest.java]: Why don't you use Arrays.asList?

File: modules/apps/headless-apio/portal/portal-apio-impl/src/test/java/com/liferay/portal/apio/internal/architect/provider/AcceptLanguageProviderTest.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+				add(Locale.US);
+			}
+		};
+
+		mockHttpServletRequest.setPreferredLocales(locales);
+
+		AcceptLanguage acceptLanguage = acceptLanguageProvider.createContext(
+			mockHttpServletRequest);
+
+		Locale locale = acceptLanguage.getPreferredLocale();
```
Comment (cgoncas) [inline modules/apps/headless-apio/portal/portal-apio-impl/src/test/java/com/liferay/portal/apio/internal/architect/provider/AcceptLanguageProviderTest.java]: Don't extract the preferredLocale to a variable if you just need it for the assert.

File: modules/apps/headless-apio/portal/portal-apio-test/src/testIntegration/java/com/liferay/portal/apio/internal/architect/provider/test/AcceptLanguageProviderTest.java [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+			new MockHttpServletRequest();
+
+		mockHttpServletRequest.setAttribute(WebKeys.USER_ID, _user.getUserId());
+
+		AcceptLanguage acceptLanguage = _provider.createContext(
+			mockHttpServletRequest);
+
+		Locale locale = acceptLanguage.getPreferredLocale();
+
+		Assert.assertNotNull(locale);
```
Comment (cgoncas) [inline modules/apps/headless-apio/portal/portal-apio-test/src/testIntegration/java/com/liferay/portal/apio/internal/architect/provider/test/AcceptLanguageProviderTest.java]: You don't need the not null assertor, don't extract the variable

File: modules/apps/headless-apio/portal/portal-apio-test/src/testIntegration/java/com/liferay/portal/apio/internal/architect/provider/test/AcceptLanguageProviderTest.java [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+		Assert.assertEquals(_USER_DEFAULT_LOCALE, locale);
+	}
+
+	private static final String _SITE_NAME =
```
Comment (cgoncas) [inline modules/apps/headless-apio/portal/portal-apio-test/src/testIntegration/java/com/liferay/portal/apio/internal/architect/provider/test/AcceptLanguageProviderTest.java]: You never check it, so use Random instead

File: modules/apps/headless-apio/portal/portal-apio-test/src/testIntegration/java/com/liferay/portal/apio/internal/architect/provider/test/AcceptLanguageProviderTest.java [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+		mockHttpServletRequest.setAttribute(WebKeys.USER_ID, _user.getUserId());
+
+		AcceptLanguage acceptLanguage = _provider.createContext(
+			mockHttpServletRequest);
+
+		Locale locale = acceptLanguage.getPreferredLocale();
+
+		Assert.assertNotNull(locale);
+
+		Assert.assertEquals(_USER_DEFAULT_LOCALE, locale);
```
Comment (cgoncas) [inline modules/apps/headless-apio/portal/portal-apio-test/src/testIntegration/java/com/liferay/portal/apio/internal/architect/provider/test/AcceptLanguageProviderTest.java]: don't use a contant, check the user default locale instead.

File: modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java [hunk 9]
```diff
@@ -522,19 +522,32 @@ private String _getLink(StructuredContentField structuredContentField) {
 		);
 	}
 
+	private Locale _getLocale(
+			AcceptLanguage acceptLanguage, long contentSpaceId)
+		throws PortalException {
+
+		Locale preferredLocale = acceptLanguage.getPreferredLocale();
+
+		if (preferredLocale != null) {
```
Comment (cgoncas) [inline modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java]: why don't you use optional?

File: modules/apps/headless-apio/structured-content/structured-content-apio-test/src/testIntegration/java/com/liferay/structured/content/apio/internal/architect/resource/test/StructuredContentNestedCollectionResourceLanguageTest.java [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+import org.junit.runner.RunWith;
+
+/**
+ * @author Ruben Pulido
+ */
+@RunWith(Arquillian.class)
+public class StructuredContentNestedCollectionResourceLanguageTest
+	extends BaseStructuredContentNestedCollectionResourceTestCase {
+
+	public static final String SITE_NAME =
```
Comment (cgoncas) [inline modules/apps/headless-apio/structured-content/structured-content-apio-test/src/testIntegration/java/com/liferay/structured/content/apio/internal/architect/resource/test/StructuredContentNestedCollectionResourceLanguageTest.java]: You never check SITE_NAME, use random instead

File: modules/apps/headless-apio/structured-content/structured-content-apio-test/src/testIntegration/java/com/liferay/structured/content/apio/internal/architect/resource/test/StructuredContentNestedCollectionResourceLanguageTest.java [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+	@Test
+	public void testGetLocaleWithNullPreferredLocale() throws Exception {
+		AcceptLanguage acceptLanguage = () -> null;
+
+		Locale locale = getLocale(acceptLanguage, _group.getGroupId());
+
+		Assert.assertEquals(_SITE_DEFAULT_LOCALE, locale);
+	}
+
+	private static final Locale _SITE_DEFAULT_LOCALE = Locale.FRANCE;
```
Comment (cgoncas) [inline modules/apps/headless-apio/structured-content/structured-content-apio-test/src/testIntegration/java/com/liferay/structured/content/apio/internal/architect/resource/test/StructuredContentNestedCollectionResourceLanguageTest.java]: Don't use contant, use the site default locale instead

**Comment by ruben-pulido** [comment]
Thank you @cgoncas! Feedback incorporated [here](https://github.com/cgoncas/liferay-portal/pull/313)

**Pull request closed without merging**